### DISCUSSION TITLE
shell-init: error after brew install

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#2554] [Getting started] [answered]
hi,

i'm on a clean macOS 11.6.1 and try to install xcode, homebrew and some apps by an "install.sh" script.
 
the first lines are:
```#!/bin/bash

xcode-select --install
read -rsp $'\n\nWait until xcode is installed and press any key...\n' -n1 key
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
read -rsp $'\n\nhomebrew done\n' -n1 key
brew upgrade 
```

at the line `brew upgrade` i got an error.

```
shell-init: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
job-working-directory: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
pwd: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
Error: The current working directory doesn't exist, cannot proceed.
```

i try a lot, also re-install macOS, but nothing helps.
the only way to fix this is to close the terminal and run the script again.

this problem is very new to me. this script works since years.
the only thing i changed is the old ruby install updated to the "new" bash script.

could someone help me please?

**Comment by gromgit** [comment, marked as answer]
>```
>shell-init: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
>job-working-directory: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
>pwd: error retrieving current directory: getcwd: cannot access parent directories: Operation not permitted
>Error: The current working directory doesn't exist, cannot proceed.
>```

Your script seems to be running in a directory that's being removed at some point before `brew upgrade`. Add `cd ~` before `brew upgrade` and see if it makes a difference.

**Reply by bluederrick** [reply]
thanks for your answer , it worked fine

**Reply by gantvi01** [reply]
cd ~ is the key
